shell/src/shell.cpp

```diff
@@ -0,0 +1,36 @@
+#include "shell/src/shell.h"
+
+namespace shell {
+
+void Shell::start() {
+
+    status_ = ShellStatus::RUNNING;
+    State state;
+    while (is_running()) {
+        std::string cmd_line;
+        std::getline(streams_->get_in_stream(), cmd_line);
+        if (cmd_line.empty()) {
+            this->stop();
+            return;
+        }
+
+        parser::Parser cmd_parser{cmd_line};
+
+        StreamsBuffered streams_local;
+
+        while (true) {
```

Comment:
It is more clear when the certain condition is checked inside while. Previous empty() method in parses looks normally